scalyr_agent/scalyr_client.py

```diff
@@ -1531,22 +1534,28 @@ def __set_attributes(self, thread_id, attributes):
 
         self.__serialization_base = tmp_buffer.getvalue()
 
-    def add_missing_attributes(self, attributes):
+    def add_missing_attributes(self, attributes, log_line_attributes=False):
         """ Adds items attributes to the base_event's attributes if the base_event doesn't
         already have those attributes set
         """
         if self.__attrs:
             changed = False
-            new_attrs = dict(self.__attrs)
+            new_attrs = dict()
+            if not log_line_attributes:
+                new_attrs = dict(self.__attrs)
             for key, value in six.iteritems(attributes):
-                if not key in new_attrs:
+                if not key in self.__attrs:
```

Comment:
As mentioned above, I think we need to also update `self.__attrs` if the value in the `self.__attrs` is not the same as what is in `attributes`.  Maybe we can make that behavior contingent on `log_line_attributes=True` in case this break existing behavior.  though, if we do that, we should have a TODO to investigate if that is really necessary.

Reply:
`self.__attrs` is set to `attributes` in the followup `__set_attributes` call, I already put in a case so that in the new format for log line attributes we don't set it but the way we currently use it it shouldn't matter if we do or don't. That field is only used when inheriting things from a base event or checking which attributes are missing during `add_missing_attributes`, neither of which we call more than once on any event as far as I can tell.